### PR TITLE
feat: add Reversed method to graph

### PR DESCRIPTION
**Pull request by miparnisari**
## Description

Add ability to visualize and represent graphs like this (the direction of a Check API request)

image

in addition to this (the direction of a ListObjects API request)

image

### CODE REVIEW

File: pkg/go/graph/graph_test.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+						define owner: [user]
+						define approved_member: member or owner
+				type group
+					relations
+						define approved_member: [user]
+				type license
+					relations
+						define active_member: approved_member from owner
+						define owner: [company, group]`,
+			expectedOutput: `digraph {
```
Comment (adriantam) [inline pkg/go/graph/graph_test.go:35]: Does it make sense to have a compare of with and without Reverse - we don't have the dot diagram in the unit test for the above model without reverse

Comment (rhamzeh) [reply pkg/go/graph/graph_test.go:35]: Maybe we can also have `authorization-model.reverse.dot` for tests, same as we have 
https://github.com/openfga/language/blob/main/tests/data/transformer/02-type-single-relation/authorization-model.dot ?

Comment (miparnisari) [reply pkg/go/graph/graph_test.go:35]: I thought about that, but it's soooo much work and i'm lazy 😆  would you be OK if in the existing tests i assert that `dot(reverse(reverse(graph)) == dot` ?

Comment (miparnisari) [reply pkg/go/graph/graph_test.go:35]: Updated @rhamzeh @adriantam

**Review by adriantam — APPROVED** [review]

**Pull request merged**